lib/config/error-codes/throw.js

```diff
@@ -67,39 +68,63 @@ function throwError (domain, subdomain, error, ...placeholders) {
   throw getError(domain, subdomain, error, ...placeholders);
 }
 
+
 /**
- * Construct and return a promise who rejects with the corresponding error
- *
+ * Construct and return the corresponding error
+ * @param  {string} errorSource - non KuzzleError from which the KuzzleError will derivate
  * @param  {string} domain - Domain (eg: 'external')
  * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
  * @param  {string} error - Error name: (eg: 'index_not_found')
  * @param  {...any} placeholders - Placeholders value to inject in error message
  */
-function rejectError (domain, subdomain, error, ...placeholders) {
-  return Promise.reject(getError(domain, subdomain, error, ...placeholders));
+function getErrorFrom (errorSource, domain, subdomain, error, ...placeholders) {
+  const derivedError = getError(domain, subdomain, error, ...placeholders);
+  
+  derivedError.message = format(derivedError.message, ...placeholders, errorSource.message || errorSource);
+  derivedError.stack = errorSource.stack;
+
+  return derivedError;
 }
 
 /**
- * Wrap getError and throw with the provided domain and subdomain.
+ * Construct and throw the corresponding error
+ * @param  {string} errorSource - non KuzzleError from which the KuzzleError will derivate
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function throwFrom (errorSource, domain, subdomain, error, ...placeholders) {
+  throw getErrorFrom(errorSource, domain, subdomain, error, ...placeholders);
+}
 
- * The function have now the following signature:
+/**
+ * Wrap getError, getErrorFrom, throw and throwFrom
+ * with the provided domain and subdomain.
+
+ * The functions have now the following signature:
  *   - getError(error, ...placeholders)
  *   - throw(error, ...placeholders)
- *
+ *   - getErrorFrom: (errorSource, error, ...placeholders)
+ *   - throwFrom: (errorSource, error, ...placeholders)
+ 
+ * @param  {string} errorSource - non KuzzleError from which the KuzzleError will derivate
  * @param  {string} domain - Domain (eg: 'external')
  * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
  */
 function wrap (domain, subdomain) {
   return {
     getError: (error, ...placeholders) => getError(domain, subdomain, error, ...placeholders),
     throw: (error, ...placeholders) => throwError(domain, subdomain, error, ...placeholders),
-    reject: (error, ...placeholders) => rejectError(domain, subdomain, error, ...placeholders)
```

Comment:
Why removing the `reject` method?

Reply:
hmm I assume its a mistake when I rebased.